lib/insights.ts

```diff
@@ -79,6 +78,8 @@ class AlgoliaAnalytics {
   public getUserToken: typeof getUserToken;
   public onUserTokenChange: typeof onUserTokenChange;
 
+  public sendEvents: ReturnType<typeof makeSendEvents>;
```

Comment:
Not blocking at all and it can be done in an other PR but it would be great if we could make the typing of the public methods more explicit so that users can see right away what is actually expected instead of a cryptic `ReturnType<typeof somePrivateMethod>`.

Reply:
Not sure if I follow you exactly.
<img width="662" alt="Screenshot 2021-11-18 at 12 03 18@2x" src="https://user-images.githubusercontent.com/499898/142403682-452c9251-f519-470d-aa9d-88f5fee3be88.png">

End-users are seeing suggestions for each method. Or, are you talking about maintainers?

Reply:
Actually, I mostly meant that for the sendEvents method mostly (because I don't think that vscode is smart enough to resolve ReturnType<typeof makeSendEvents>, but it seems it is not exposed in the codesandbox.

Digging into the issue, I realised that you should also add the method in the InsightsMethodMap type in types.ts if you want it to appear in the types for `aa('sendEvents', [...])`

Reply:
vscode seems smart enough to resolve ReturnType, but yes I missed adding it to `InsightsMethodMap`.
<img width="891" alt="Screenshot 2021-11-18 at 13 25 32@2x" src="https://user-images.githubusercontent.com/499898/142415132-e67f5144-24e8-4ab0-893b-16dcd052f775.png">

https://github.com/algolia/search-insights.js/pull/335/commits/f0daa73d3429d03d9c738925aee922d9c7a6ad8c

Reply:
Yup that's awesome 🙂